modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -2,18 +2,52 @@
 
 #include "backends/common/serialization.hpp"
 
+namespace {
+    struct MyCustomType {
+        int val;
+        std::string name;
+        std::vector<float> vec;
+        std::map<int, uint64_t> mmap;
+        bool operator==(const MyCustomType& other) const {
+            return val == other.val && name == other.name &&
+                   vec == other.vec && mmap == other.mmap;
+        }
+    };
+}
+
+// FIXME: namespace conflicts
+namespace cv {
+namespace gapi {
+namespace s11n {
+namespace detail {
```

Comment:
This is way to long, need to think how to make declaring this `S11N<>` easier for users. Maybe with a macro like we do with `G_API_OP`, etc?

Reply:
Can be done separately

Reply:
Please open a task for tat